### PR TITLE
🎯 allow manual specification of speaker number

### PR DESCRIPTION
**Pull request by anuejn**

### CODE REVIEW

File: backend/openapi-schema.yml [hunk 1]
```diff
@@ -85,6 +85,9 @@ components:
         model:
           title: Model
           type: string
+        n_speakers:
```
Comment (phlmn) [inline backend/openapi-schema.yml:88]: Could be a more descriptive name

File: frontend/src/pages/new_document.tsx [hunk 2]
```diff
@@ -69,12 +74,20 @@ export function NewDocumentPage() {
 
     try {
       setLoading(true);
-      const response = await createDocument({
+
+      const documentParameters: Parameters<typeof createDocument>[0] = {
```
Comment (pajowu) [inline frontend/src/pages/new_document.tsx:78]: Maybe extract to type defintion and then use that here?

File: frontend/src/pages/new_document.tsx [hunk 3]
```diff
@@ -230,11 +247,74 @@ export function NewDocumentPage() {
...
+                    'border-black dark:border-white border-2 rounded-r',
+                  )}
+                >
+                  advanced
+                </label>
+              </div>
+            </FormControl>
+            {speakerDetection == 'advanced' && (
+              <FormControl label="number of speakers" className="-mt-4">
+                <Input type="number" {...register('numberOfSpeakers')} />
```
Comment (pajowu) [inline frontend/src/pages/new_document.tsx:314]: probably needs minimal value?

File: frontend/src/pages/new_document.tsx [hunk 4]
```diff
@@ -134,7 +147,7 @@ export function NewDocumentPage() {
                 <div
                   className={clsx(
                     'absolute',
-                    'top-1',
+                    'topy-2 text-center',
```
Comment (phlmn) [inline frontend/src/pages/new_document.tsx:150]: typo? `topy-2` -> `top-2`

File: frontend/src/pages/new_document.tsx [hunk 5]
```diff
@@ -230,11 +247,74 @@ export function NewDocumentPage() {
...
+              </div>
             ) : (
               <></>
             )}
 
+            <FormControl label={'speaker detection'}>
+              <div className="flex">
+                <input
+                  type="radio"
+                  id="off"
```
Comment (phlmn) [inline frontend/src/pages/new_document.tsx:259]: pretty common ids... maybe autogenerate?

File: backend/transcribee_backend/routers/document.py [hunk 6]
```diff
@@ -129,20 +133,22 @@ def create_default_tasks_for_document(
...
+        )
+        session.add(speaker_identification_task)
 
 
 @document_router.post("/")
 async def create_document(
     name: str = Form(),
     model: str = Form(),
     language: str = Form(),
+    n_speakers: Optional[int] = Form(None),
```
Comment (pajowu) [inline backend/transcribee_backend/routers/document.py:151]: The validation could also be added here https://fastapi.tiangolo.com/tutorial/path-params-numeric-validations/#number-validations-greater-than-or-equal

**Review by pajowu — CHANGES_REQUESTED** [review]

**Pull request merged**